src/config.ts

```diff
@@ -285,6 +301,23 @@ const defaultConfig: BaseConfig = {
   setPendingStatusCheck: false,
 };
 
+const defaultGenerateOnlyConfig: ProjectConfig = {
+  lostPixelProjectId: '---',
+  ciBuildId: process.env.GITHUB_RUN_ID ?? '1',
+  ciBuildNumber: process.env.GITHUB_RUN_NUMBER ?? '1',
+  repository: process.env.REPOSITORY ?? 'unknown/unknown',
+  commitRef: process.env.GITHUB_REF ?? 'main',
+  commitRefName: process.env.GITHUB_REF_NAME ?? 'main',
+  commitHash:
+    process.env.COMMIT_HASH ?? '0000000000000000000000000000000000000000',
+  s3: {
+    accessKey: '---',
+    secretKey: '---',
+    bucketName: '---',
+    endPoint: 'https://s3.amazonaws.com',
+  },
+};
```

Comment:
why do we even need this in the default generate only config?  looks like we can totally omit this, no?

Reply:
it's a bit complicated.
on the one hand, we want to have these as required for the platform, on the other, we want to omit them in `generateOnly` mode.
I believe it's safer to keep them required and work with sane defaults, rather than make them optional, and later with new additions to forget about it.

Reply:
But you still removed them from generate templates. How that would work?

Reply:
treating config (type) differently depending on the `generateOnly` flag now.